StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1140MaximumLineLength.cs

```diff
@@ -0,0 +1,73 @@
+    /// </summary>
+    /// <remarks>
+    /// <para>A violation of this rule occurs whenever the code contains lines with more characters
+    /// than allowed in stylecop.json</para>
+    /// </remarks>
+    [NoCodeFix("No code fix implemented")]
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal class SA1140MaximumLineLength : DiagnosticAnalyzer
+    {
+        /// <summary>
+        /// The ID for diagnostics produced by the <see cref="SA1140MaximumLineLength"/> analyzer.
+        /// </summary>
+        public const string DiagnosticId = "SA1140";
+        private static readonly LocalizableString Title = new LocalizableResourceString(nameof(ReadabilityResources.SA1140Title), ReadabilityResources.ResourceManager, typeof(ReadabilityResources));
+        private static readonly LocalizableString MessageFormat = new LocalizableResourceString(nameof(ReadabilityResources.SA1140MessageFormat), ReadabilityResources.ResourceManager, typeof(ReadabilityResources));
+        private static readonly LocalizableString Description = new LocalizableResourceString(nameof(ReadabilityResources.SA1140Description), ReadabilityResources.ResourceManager, typeof(ReadabilityResources));
+        private static readonly string HelpLink = "https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1140.md";
+
+        private static readonly DiagnosticDescriptor Descriptor =
+            new DiagnosticDescriptor(DiagnosticId, Title, MessageFormat, AnalyzerCategory.ReadabilityRules, DiagnosticSeverity.Warning, AnalyzerConstants.EnabledByDefault, Description, HelpLink);
+
+        private static readonly Action<SyntaxTreeAnalysisContext> SyntaxTreeAction = HandleSyntaxTree;
+
+        /// <inheritdoc/>
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } =
+            ImmutableArray.Create(Descriptor);
+
+        /// <inheritdoc/>
+        public override void Initialize(AnalysisContext context)
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.EnableConcurrentExecution();
+
+            context.RegisterSyntaxTreeAction(SyntaxTreeAction);
+        }
+
+        private static void HandleSyntaxTree(SyntaxTreeAnalysisContext context)
+        {
+            var readabilityRules = context.GetStyleCopSettings(context.CancellationToken).ReadabilityRules;
+
+            // Only proceed if rule is enabled
+            if (readabilityRules.MaximumLineLength > 0)
+            {
+                SyntaxNode root = context.Tree.GetCompilationUnitRoot(context.CancellationToken);
+                var fileText = context.Tree.GetText(context.CancellationToken);
+                foreach (var line in fileText.Lines)
+                {
+                    if ((line.End - line.Start) > readabilityRules.MaximumLineLength)
+                    {
+                        var location = root.SyntaxTree.GetLocation(line.Span);
```

Comment:
Maybe only report the diagnostic on the bit that is too long so the user knows how much the line is too long.